src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java

```diff
@@ -596,50 +434,33 @@ else if (buildingToDeliver instanceof TileEntityColonyBuilding)
      */
     private IAIState prepareDelivery()
     {
-        final IBuildingWorker ownBuilding = getOwnBuilding();
-        if (ownBuilding instanceof BuildingDeliveryman)
+        final IRequest<? extends IRequestable> currentTask = job.getCurrentTask();
+        final IRequestable request = currentTask.getRequest();
+        if (!(request instanceof Delivery))
         {
-            final IRequest<? extends Delivery> request = job.getCurrentTask();
-            if (request != null)
-            {
-                if (job.isReturning())
-                {
-                    return DUMPING;
-                }
-                ((IBuildingDeliveryman) ownBuilding).setBuildingToDeliver(request.getRequest().getTarget());
-                if (InventoryUtils.hasItemInItemHandler(worker.getInventoryCitizen(),
-                  itemStack -> request.getRequest().getStack().isItemEqualIgnoreDurability(itemStack)))
-                {
-                    return DELIVERY;
-                }
+            // The current task has changed since the Decision-state.
+            // Restart.
+            return START_WORKING;
+        }
 
-                return gatherItems(request);
-            }
+        final Delivery delivery = (Delivery) request;
+        if (InventoryUtils.hasItemInItemHandler(worker.getInventoryCitizen(),
+          itemStack -> delivery.getStack().isItemEqualIgnoreDurability(itemStack)))
```

Comment:
think you want to use an ItemStackUtils comparing function for this since afaik requests are also nbt sensitive

Reply:
Blame @Raycoms. I only changed the formatting here, not the logic. Ray, what do you think?

Reply:
interesting, maybe thats one of the things throwing the dman off

Reply:
Since he got the exact delivery from the RS, just equals should be fine here yes. Maybe having one that matches with ignore durability but doesn't match the normal way here, is really throwing him off.

Reply:
isItemEqual I mean.

Reply:
He need to check for nbt and meta equal. Item Equality is not enough at this stage.

Reply:
Yes, isItemEqual does check that. While isItemEqualIgnoreDurability checks less.

Reply:
Okey yeah then it needs to be isItemEqual

Reply:
Changed in 51e3213.

Reply:
Okay, tried it one more time with aac342d.
It now uses `ItemStackUtils.compareItemStacksIgnoreStackSize`, same as `gatherIfInTileEntity`.